launch/lifelong_launch.py

```diff
@@ -1,17 +1,51 @@
-from launch import LaunchDescription
-import launch_ros.actions
 from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import EmitEvent, LogInfo, RegisterEventHandler
+from launch.events import matches_action
+from launch_ros.actions import LifecycleNode
+from launch_ros.event_handlers import OnStateTransition
+from launch_ros.events.lifecycle import ChangeState
+from lifecycle_msgs.msg import Transition
 
 
 def generate_launch_description():
-    return LaunchDescription([
-        launch_ros.actions.Node(
+    start_lifelong_slam_toolbox_node = LifecycleNode(
           parameters=[
             get_package_share_directory("slam_toolbox") + '/config/mapper_params_lifelong.yaml'
           ],
           package='slam_toolbox',
           executable='lifelong_slam_toolbox_node',
           name='slam_toolbox',
-          output='screen'
+          output='screen',
+          namespace='/'
+    )
+
+    configure_event = EmitEvent(
+        event=ChangeState(
+            lifecycle_node_matcher=matches_action(start_lifelong_slam_toolbox_node),
+            transition_id=Transition.TRANSITION_CONFIGURE
+        )
+    )
+
+    activate_event = RegisterEventHandler(
+        OnStateTransition(
+            target_lifecycle_node=start_lifelong_slam_toolbox_node,
+            start_state="configuring",
+            goal_state="inactive",
```

Comment:
Goal state is inactive? That doesn't seem like it would activate things

Reply:
Ditto here and other launch files

Reply:
OnStateTransition is the EventHandler. After detecting the `configuring -> inactive` transition of the target lifecycle node, The EmitEvent(`Transition.TRANSITION_ACTIVATE`) is processed.

ros2 official example: <https://github.com/ros2/launch_ros/blob/humble/launch_ros/examples/lifecycle_pub_sub_launch.py#L42-L55>

---

![image](https://user-images.githubusercontent.com/32297845/215664765-c28a6910-756c-448b-8ed2-3258d5164669.png)